Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -30,85 +27,240 @@ export default function widgetBehavior(publicAPI, model) {
+    }
+  }
+
+  function updateHandleDirection(nb) {
+    const direction = orientHandlesWithCompleteWidget(nb);
+    orientHandle(nb, direction);
+  }
+
+  publicAPI.updateHandleDirections = () => {
+    if (isHandleOrientable(handle1.getShape())) {
+      updateHandleDirection(1);
+    }
+    if (isHandleOrientable(handle2.getShape())) {
+      updateHandleDirection(2);
+    }
+  };
+
+  publicAPI.rotateHandlesToFaceCamera = () => {
+    if (
+      (model.representations[0].getFaceCamera() === FaceCameraStates.NONE &&
+        publicAPI.isHandleGlyph2D(handle1.getShape())) ||
+      model.representations[0].getFaceCamera() === FaceCameraStates.TRUE
+    ) {
+      model.representations[0].setViewMatrix(
+        Array.from(model.camera.getViewMatrix())
+      );
+    }
+    if (
+      (model.representations[1].getFaceCamera() === FaceCameraStates.NONE &&
+        publicAPI.isHandleGlyph2D(handle2.getShape())) ||
+      model.representations[1].getFaceCamera() === FaceCameraStates.TRUE
+    ) {
+      model.representations[1].setViewMatrix(
+        Array.from(model.camera.getViewMatrix())
+      );
+    }
+  };
+
+  // Handles visibility ---------------------------------------------------------
+
+  publicAPI.activateHandleVisibility = () => {
+    if (handle1.getVisible() === false && handle1.getActive()) {
+      handle1.setVisible(true);
+      handleToHide[0] = true;
+    }
+    if (handle2.getVisible() === false && handle2.getActive()) {
+      handle2.setVisible(true);
+      handleToHide[1] = true;
+    }
+  };
+
+  publicAPI.deactivateHandleVisibility = () => {
+    if (handleToHide[0] === true) {
+      handleToHide[0] = false;
+      handle1.setVisible(false);
+    }
+    if (handleToHide[1] === true) {
+      handleToHide[1] = false;
+      handle2.setVisible(false);
     }
   };
 
+  publicAPI.toggleMoveHandleVisibility = (visibility) => {
+    model.representations[2].setVisibilityFlagArray([visibility, visibility]);
+    model.widgetState.getMoveHandle().setVisible(visibility);
+    model.representations[2].updateActorVisibility();
+  };
+
+  /*
+   * Set actor visibility to true unless it is a NONE handle
+   * and uses state visibility variable for the displayActor visibility to
+   * allow pickable handles even when they are not displayed on screen
+   */
+  publicAPI.manageHandleVisibility = (handle, nb) => {
+    const visibility = handle.getVisible();
+    model.representations[nb - 1].setVisibilityFlagArray([
+      visibility,
+      handle.getShape() !== 'voidSphere',
+    ]);
+    handle.setVisible(visibility);
+    model.representations[nb - 1].updateActorVisibility();
+    if (nb === 1) {
+      model.representations[nb].updateActorVisibility();
+    }
+    model.interactor.render();
+  };
+
+  // --------------------------------------------------------------------------
+
+  publicAPI.placeHandle = (handle) => {
+    const nb = getNbHandles(model);
+    publicAPI.manageHandleVisibility(handle, nb);
```

Comment:
it seems to me that nb could be computed inside `manageHandleVisibility` instead

Reply:
I would set visibility before configuring the handles

Reply:
this is what i am doing in my mind. 1st thing to be called after getting handle number data (which handle is placed) is setting visibility feature. Am i missing something here?

Reply:
set origin color, scale and text BEFORE calling manageHandleVisibility